src/mixin/pending-container-mixin.js

```diff
@@ -0,0 +1,51 @@
+
+export const PendingContainerMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			_hasPendingChildren: {
+				type: Boolean
+			}
+		};
+	}
+
+	constructor() {
+		super();
+		this.__pendingCount = 0;
```

Comment:
The `__` and `_` in Polymer always kind of confused me - is there a convention for this?

Reply:
We've had a couple of conversations about this but I feel like it's still a loose-end perhaps because folks were/are focused on other important things.  :)

[This StackOverflow question/answer](https://stackoverflow.com/questions/35946339/double-underscore-vs-single-underscore-in-nodejs) seems to indicate double underscore is not an "official" convention.  Apparently single underscore is the only "unofficial" convention for indicating "private" members, and the double underscore came about due to 2 NodeJS global variables, `__dirname` and `__filename`.

I think we've used double underscore in certain places in effort to avoid possible collision between components and mixins.  I think the different underscores have also been used by some to differentiate between protected and private - this muddies things up even more, and I don't think there is much convention for this.

My personal view is that we should use single underscore for private variables everywhere, and behaviors/mixins should name their members to best avoid such collision.  I'd rather have a funky verbose name in one mixin than force every consumer of that mixin to have an overly verbose name.

The "protected" case is interesting, and I am not sure how these should be named. i.e. when a mixin defines a member it intends the sub-class to use, but it does not expect it to be used by others.  I tend to think it should simply be treated like a public member, but I acknowledge that I prefer `_entity` to be "private", mostly because it is unsettling to me to have both `entity` and `href` as public members.

I am happy to use whatever convention we agree on - I'd rather be consistent than not, even if it's not my preference.  :)

Reply:
I see a lot of usages of `_` and `__` in some of the `brightspace-ui/core` projects. My understanding has always been that yeh `_` is protected, `__` is private, for what little value that has in JS anyway. I don't feel strongly about it either way but will probably leave as is until someone says it's categorically something we should stop doing.

Reply:
Yeah we haven't been consistent in our use in `core`.  I'd like to fix that, and in `core` it has usually been used to avoid collision between private mixin members and private component members with the same name.  Technically it doesn't provide any real runtime protection, but it helps understand the code since if infers scope.